packages/ubuntu_desktop_installer/lib/pages/allocate_disk_space/storage_types.dart

```diff
@@ -9,7 +9,67 @@ extension DiskExtension on Disk {
 }
 
 extension PartitionExtension on Partition {
-  bool get canWipe => mount != null;
+  bool get canWipe => PartitionFormat.fromPartition(this)?.canWipe == true;
   bool get isWiped => wipe == 'superblock';
   String get prettySize => filesize(size ?? 0);
 }
+
+/// Available partition formats.
+class PartitionFormat {
+  const PartitionFormat._(this.type);
+
+  /// The type of the partition format (e.g. 'ext4').
+  final String type;
+
+  /// Whether a partition with this format can be wiped.
+  bool get canWipe => type != 'swap';
```

Comment:
Could this ever be inquired about a partition containing an existing OS the user won't want to wipe out?

Reply:
https://github.com/canonical/ubuntu-desktop-installer/issues/1101